ruby/red-arrow/test/test-record-batch-raw-records.rb

```diff
@@ -0,0 +1,401 @@
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class RecordBatchRawRecordsTest < Test::Unit::TestCase
+  sub_test_case("instance methods") do
+    sub_test_case("#raw_records") do
+      sub_test_case("with basic arrays") do
+        def setup
+          @string_values = ["apple", "orange", "watermelon", nil, "タコ"]
+          @string_array = Arrow::StringArray.new(@string_values)
+
+          @uint32_values = [1, 2, 4, 8, 16]
+          @uint32_array = Arrow::UInt32Array.new(@uint32_values)
+
+          @double_values = [10.1, 11.2, 12.3, 13.4, 14.4]
+          @double_array = Arrow::DoubleArray.new(@double_values)
+
+          @decimal128_values = ['123.45', '234.56', nil, '345.67', "456.78"]
+          @decimal128_array = Arrow::Decimal128Array.new(
+            Arrow::Decimal128DataType.new(8, 2), @decimal128_values)
+
+          @date32_values = [
+            Date.new(1993,  2, 24),
+            Date.new(1996, 12, 25),
+            Date.new(2013,  2, 24),
+            Date.new(2020, 12, 25),
+            nil
+          ]
+          epoch_date = Date.new(1970, 1, 1)
+          @date32_array = Arrow::Date32Array.new(
+            @date32_values.map {|dt| dt && (dt - epoch_date).to_i }
+          )
+
+          @date64_values = [
+            nil,
+            DateTime.new(1993,  2, 23, 15, 0, 0),
+            DateTime.new(1996, 12, 24, 15, 0, 0),
+            DateTime.new(2013,  2, 23, 15, 0, 0),
+            DateTime.new(2020, 12, 24, 15, 0, 0)
+          ]
+          @date64_array = Arrow::Date64Array.new(
+            @date64_values.map {|dt| dt && dt.to_time.gmtime.to_i * 1000 }
+          )
+
+          jst = '+09:00'
+          @timestamp_values = [
+            Time.new(1993,  2, 24, 0, 0, 0, jst).gmtime,
+            Time.new(1996, 12, 25, 0, 0, 0, jst).gmtime,
+            nil,
+            Time.new(2013,  2, 24, 0, 0, 0, jst).gmtime,
+            Time.new(2020, 12, 25, 0, 0, 0, jst).gmtime
+          ]
+          @timestamp_sec_array = Arrow::TimestampArray.new(
+            :second,
+            @timestamp_values.map {|ts| ts && ts.to_i }
+          )
+          @timestamp_msec_array = Arrow::TimestampArray.new(
+            :milli,
+            @timestamp_values.map {|ts| ts && (ts.to_r * 1_000).to_i }
+          )
+          @timestamp_usec_array = Arrow::TimestampArray.new(
+            :micro,
+            @timestamp_values.map {|ts| ts && (ts.to_r * 1_000_000).to_i }
+          )
+
+          @dict_vocab = Arrow::StringArray.new(['foo', 'bar', 'baz'])
```

Comment:
Could you use `@dictionary` because Arrow C++ and C GLib use `dictionary`.

Reply:
OK.  But I think `vocabulary` is a better word than `dictionary` for this array.

Reply:
I reconsidered that `dictionary` is also suitable in this case as I remembered we say `dictionary` in the similar data structure in LZ77 algorithm.

Reply:
Renamed.